src/gatsby/hooks/useClickOutside.tsx

```diff
@@ -0,0 +1,23 @@
+import { useEffect } from 'react'
```

Comment:
Q: where do we use this hook, should it be placed locally? (vs globally into gatsby/hooks)

Reply:
This hook is to detect when we click outside a certain element block. Most use cases are on Modal, dropdowns, etc. where we want to close them when we click outside of that. This hook is reusable and often used on those use cases.

Reply:
do we have already cases like this? we have quite a lot of dropdowns

Reply:
Yeah actually. I just checked the `What's new` section on our homepage. That part is also somewhat similar to implementation but is not reusable. We can actually use this reusable hook there as well and also on upcoming similar functionality.

Reply:
how about the download button? top menu drop down?

Let's create a ticket please to unify this, so that we don't duplicate and have 3 different implementations please.